dipy/core/gradients.py

```diff
@@ -210,7 +211,26 @@ def bvecs(self):
         denom = self.bvals + (self.bvals == 0)
         denom = denom.reshape((-1, 1))
         return self.gradients / denom
-
+    
+    def getitem(self, bmin=0, bmax=np.inf):
+        if bmin != self.b0_threshold:
+            self.b0_threshold = bmin
+            warn("Updating b0_threshold to {} for slicing.".format(bmin), UserWarning, stacklevel=2)
+        # priority to bmin in all other cases
+        mask = (self.bvals >= bmin) & (self.bvals <= bmax)
```

Comment:
like you did before, check if `slice` or just `int`

Reply:
I'm sorry I didn't understand. Can u elaborate on what needs to be checked?

Reply:
Were you referring to bmin value?

Reply:
Oh, sorry. I understood what you meant.